kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/marshal/MarshallerHelper.java

```diff
@@ -120,7 +123,11 @@ public String marshal(String marshallingFormat, Object entity) {
 
         Marshaller marshaller = serverMarshallers.get(format);
         if (marshaller == null) {
-            marshaller = MarshallerFactory.getMarshaller(format, this.getClass().getClassLoader());
+        	if (registry != null) {
+        		marshaller = MarshallerFactory.getMarshaller(registry.getExtraClasses(), format, this.getClass().getClassLoader());
+        	} else {
+        		marshaller = MarshallerFactory.getMarshaller(format, this.getClass().getClassLoader());
+        	}
```

Comment:
maybe this if/else to be replaced with:
```
marshaller = MarshallerFactory.getMarshaller(getExtraClasses(registry), format, this.getClass().getClassLoader());
```

and the getExtraClasses(registry) can do the check and if no registry return empty set. wdyt?